base/doc/clsguide.tex

```diff
@@ -1181,6 +1181,113 @@ \subsection{Querying localisation}
+   This command is filled beside others by the \pkg{nameref} package and some
+   classes (e.g.~\pkg{memoir}) and typically gives the title of some sectioning command.
+ \item[\texttt{target}] The content of \cs{@currentHref}. 
+   This command is normally filled by \pkg{hyperref} and gives the name of the last destination it created.  
+ \item[\texttt{counter}] The content of \cs{@currentcounter}. 
+   This command contains after a \cs{refstepcounter} the name of the counter.
+ \item[\texttt{xpos}, \texttt{ypos}] These properties records the $x$~and $y$ coordinates of a point previously 
+   stored with \cs{pdfsavepos}/\cs{savepos}. 
+   E.g.~(if \pkg{bidi} is used it can be necessary to save the position
+   before and after the label):
+   \begin{verbatim}
+     \pdfsavepos 
+     \RefProperty{myposition}{xpos,ypos}%
+     \pdfsavepos
+   \end{verbatim}  
+\end{description} 
+ 
+Class and package authors can define more properties to store other values they are interested in.
+\begin{decl}
+  |\NewProperty|\arg{name}\arg{setpoint}\arg{default}\arg{code}\\
+  |\SetProperty|\arg{name}\arg{setpoint}\arg{default}\arg{code}
+\end{decl}
+These commands declare or change a property \meta{name}\footnote{The standard set of properties and properties of other package should not be changed!}. If a new property 
+is declared within a package it is suggested that its name is always structured as follows:
+\meta{package-name}\texttt{/}\meta{property-name}. \meta{setpoint} is either |now| or |shipout| and decides if the 
+value is written directly or at the next shipout. \meta{default} is used if the property is referenced but not yet known. 
+\meta{code} is the code executed when storing the value. For example the \texttt{pagenum} property is declare as
+\begin{verbatim}
+\NewProperty{pagenum}{shipout}{0}{\the\value{page}}
+\end{verbatim}
+
+The commands related to properties are offered as a set of CamelCase
+commands for traditional \LaTeXe{} packages (and for use in the
+document preamble if needed) as well as \texttt{expl3} commands
+for modern packages, that use the L3 programming layer of
+\LaTeX{}. The \texttt{expl3} commands and more details can by found in \texttt{ltproperties-doc.pdf}. 
+
+\subsection{Preparing link targets}
+
+Active links in a document need targets where they can jump to. Such targets are often created automatically (if the package \pkg{hyperref} is loaded) by the \cs{refstepcounter} command but there are also cases where class or package authors need to add a target manually for example in unnumbered sectioning commands or in environments. For this \LaTeX{} provides the following commands. \emph{Without} \pkg{hyperref} they do nothing or insert only a whatsits, \emph{with} \pkg{hyperref} they add the necessary targets. Details about the behaviour and the arguments of the following commands can by found in the \pkg{hyperref} package in \texttt{hyperref-linktarget.pdf}. 
```

Comment:
appreciate if you put newlines into your paragraphs :-) here and elsewhere ... much better for diffing

Reply:
targets where -> target to where ??? not sure

Reply:
only a whatsit -> only a whatsit (to ensure that spacing doesn't change when hyperref is loaded),

Reply:
"to which they can jump"

Not that I approve of using the verb "jump" here!  But will leave that to my full review.